web/src/components/storage/ProposalSettingsSection.jsx

```diff
@@ -509,6 +510,162 @@ const EncryptionPasswordField = ({
   );
 };
 
+/**
+ * Form for configuring the space policy.
+ * @component
+ *
+ * @param {object} props
+ * @param {string} props.id - Form ID.
+ * @param {ProposalSettings} props.settings - Settings used for calculating a proposal.
+ * @param {onSubmitFn} [props.onSubmit=noop] - On submit callback.
+ *
+ * @callback onSubmitFn
+ * @param {string} policy - Name of the selected policy.
+ */
+const SpacePolicySettingsForm = ({
+  id,
+  settings,
+  onSubmit: onSubmitProp = noop
+}) => {
+  const [spacePolicy, setSpacePolicy] = useState(settings);
+
+  const selectDelete = () => {
+    setSpacePolicy("delete");
+  };
+
+  const selectResize = () => {
+    setSpacePolicy("resize");
+  };
+
+  const selectKeep = () => {
+    setSpacePolicy("keep");
+  };
+
+  const onSubmit = (e) => {
+    e.preventDefault();
+    onSubmitProp(spacePolicy);
+  };
+
+  return (
+    <Form id={id} onSubmit={onSubmit}>
+      <div className="split">
+        <span>{_("Policy to make enough free space")}</span>
+        <ToggleGroup isCompact>
+          <ToggleGroupItem
+            text={_("Delete Anything")}
+            buttonId="deletePolicy"
+            isSelected={spacePolicy === "delete"}
+            onClick={selectDelete}
+          />
+          <ToggleGroupItem
+            text={_("Resize Only")}
```

Comment:
```suggestion
            text={_("Shrink Only")}
```

Help text

> Some of the existing partitions will be resized to make them smaller, keeping their contents, just enough to make space for the system being installed.

I wonder, does this really work automatically? Last time I used something like this, I wanted to specify **how much** space would be allocated to Linux and how much the fresh and mostly empty Windows system would get to keep.